ui/cypress/e2e/collectors.test.ts

```diff
@@ -138,52 +147,76 @@ describe('Collectors', () => {
-
-        cy.getByTestID('search-widget')
-          .clear()
-          .type('a')
-
-        cy.getByTestID('resource-card').should('have.length', 2)
-        cy.getByTestID('resource-card').should('contain', firstTelegraf)
-        cy.getByTestID('resource-card').should('contain', secondTelegraf)
-        cy.getByTestID('resource-card').should('not.contain', thirdTelegraf)
-      })
-      // sort by buckets test here
-      it('can sort telegraf configs by bucket', () => {
-        cy.getByTestID('bucket-name').should('have.length', 3)
-        cy.getByTestID('bucket-sorter').click()
+          .type(firstTelegraf)
+          .then(() => {
+            cy.getByTestID('resource-card').should('have.length', 1)
+            cy.getByTestID('resource-card').should('contain', firstTelegraf)
+
+            cy.getByTestID('search-widget')
+              .clear()
+              .type(secondTelegraf)
+              .then(() => {
+                cy.getByTestID('resource-card').should('have.length', 1)
+                cy.getByTestID('resource-card').should(
+                  'contain',
+                  secondTelegraf
+                )
+
+                cy.getByTestID('search-widget')
+                  .clear()
+                  .type(thirdTelegraf)
+                  .then(() => {
+                    cy.getByTestID('resource-card').should('have.length', 1)
+                    cy.getByTestID('resource-card').should(
+                      'contain',
+                      thirdTelegraf
+                    )
+
+                    cy.getByTestID('search-widget')
+                      .clear()
+                      .type('should have no results')
+                      .then(() => {
+                        cy.getByTestID('resource-card').should('have.length', 0)
+                        cy.getByTestID('empty-state').should('exist')
+
+                        cy.getByTestID('search-widget')
+                          .clear()
+                          .type('a')
+                          .then(() => {
+                            cy.getByTestID('resource-card').should(
+                              'have.length',
+                              2
+                            )
+                            cy.getByTestID('resource-card').should(
+                              'contain',
+                              firstTelegraf
+                            )
+                            cy.getByTestID('resource-card').should(
+                              'contain',
+                              secondTelegraf
+                            )
+                            cy.getByTestID('resource-card').should(
+                              'not.contain',
+                              thirdTelegraf
+                            )
+                          })
+                      })
+                  })
+              })
+          })
 
-        bucketz.sort()
+        // sort by buckets test here
+        cy.reload()
```

Comment:
why reload here?

Reply:
it's more a conceptual reload than anything. just kind of like a break in the last bit of tests. i'll remove them

Reply:
so i smooshed three separate specs into a single spec to limit teardown and rebuilding and such, but a consequence of that these is that there is some wonky leftover state from the previous test that affects the current test (like active filtering or sorting for example). these `reload`s are necessary to clear out that state